pkg/controller/registry/resolver/resolver.go

```diff
@@ -531,12 +531,16 @@ func (r *SatResolver) sortChannel(bundles []*Operator) ([]*Operator, error) {
 	}
 
 	for _, b := range bundles {
-		if b.replaces == "" {
-			continue
+		if b.replaces != "" {
+			if r, ok := bundleLookup[b.replaces]; ok {
+				replacedBy[r] = b
+				replaces[b] = r
+			}
 		}
-		if r, ok := bundleLookup[b.replaces]; ok {
-			replacedBy[r] = b
-			replaces[b] = r
+		for _, skip := range b.skips {
+			if r, ok := bundleLookup[skip]; ok {
+				replacedBy[r] = b
```

Comment:
in other places that we have skips (i.e. the graphloader), this becomes an array.

`replaces[*Operator][]*Operator`

so `replaces[name][0]` is `replaces` and `replaces[name][1..]` is `skips`

Reply:
We use `replacedBy` to just keep a record of which versions are replaced by either `replaces` or `skips`. This map is used to determine the head of the channel so the key is used, not the value. There should be a version that doesn't get replaced anything and that is the head. 
The `replaces[]` is used to determine the order which we consider the versions that are replaced, not skipped because if they are skipped, they should be ignored in eligible versions for installation. Hence I don't add skipped versions to `replaces[]`. I added the `skipped[]` to simply an insurance that will filter some stray skipped versions.